File: mater-webapp/src/app/store/dtos/user.dto.ts
import {RoleDto} from "./role.dto";
import {PersonalDto} from "./personal.dto";
import {AbstractDto} from "./core/abstract.dto";
import {SellerDto} from "./seller.dto";
import {UserRegistrationStateEnum} from "../enums/user-registration-state.enum";

export class UserDto extends AbstractDto {
  username: string;
  isUsernameSet: boolean;
  password: string;
  email: string;
  isEmailSet: boolean;
  personal: PersonalDto;
  seller: SellerDto;
  telegramId: number;
  telegramChatId: number;
  registeredAt: Date;
  status: UserRegistrationStateEnum;
  roles: RoleDto[];
}

File: mater-rest/src/main/java/com/oneune/mater/rest/main/store/dtos/UserDto.java
package com.oneune.mater.rest.main.store.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.oneune.mater.rest.main.store.dtos.core.AbstractDto;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;

import java.time.Instant;
import java.util.List;

/**
 * @see com.oneune.mater.rest.main.store.entities.UserEntity
 */
@FieldDefaults(level = AccessLevel.PRIVATE)
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Data
public class UserDto extends AbstractDto {
    String username;
    @JsonProperty("isUsernameSet")
    boolean isUsernameSet;
    String email;
    @JsonProperty("isEmailSet")
    boolean isEmailSet;
    PersonalDto personal;
    SellerDto seller;
    boolean registeredByTelegram;
    Long telegramId;
    Long telegramChatId;
    Instant registeredAt;
    List<RoleDto> roles;
}
